Mathlib/SetTheory/Ordinal/Topology.lean

```diff
@@ -301,9 +301,82 @@ theorem IsAcc.subset {o : Ordinal} {S T : Set Ordinal} (h : S ⊆ T) (ho : o.IsA
+  constructor
+  · intro h
+    have plim : p.IsLimit := IsAcc.isLimit h
+    rw [accPt_iff_nhds] at *
+    intro u hu
+    obtain ⟨l, hl⟩ := exists_Ioc_subset_of_mem_nhds hu ⟨⟨0, plim.pos.trans hpo⟩, plim.pos⟩
+    obtain ⟨x, hx⟩ := h (Ioo l (p + 1)) (Ioo_mem_nhds hl.1 (lt_add_one _))
+    use ⟨x, lt_of_le_of_lt (lt_succ_iff.mp hx.1.1.2) hpo⟩
+    refine ⟨?_, Subtype.coe_ne_coe.mp hx.2⟩
+    exact ⟨hl.2 ⟨hx.1.1.1, by exact_mod_cast lt_succ_iff.mp hx.1.1.2⟩, hx.1.2⟩
+  · intro h
+    rw [accPt_iff_nhds] at *
+    intro u hu
+    by_cases ho : p + 1 < o
+    · have ppos : p ≠ 0 := by
+        rintro rfl
+        rw [zero_add] at ho
+        specialize h (Iio ⟨1, ho⟩) (Iio_mem_nhds (Subtype.mk_lt_mk.mpr zero_lt_one))
+        obtain ⟨_, h⟩ := h
+        exact h.2 <| Subtype.mk_eq_mk.mpr (lt_one_iff_zero.mp h.1.1)
+      have plim : p.IsLimit := by
+        contrapose! h
+        obtain ⟨q, hq⟩ := ((zero_or_succ_or_limit p).resolve_left ppos).resolve_right h
+        use (Ioo ⟨q, ((hq ▸ lt_succ q).trans hpo)⟩ ⟨p + 1, ho⟩)
+        constructor
+        · exact Ioo_mem_nhds (by simp only [hq, Subtype.mk_lt_mk, lt_succ]) (lt_succ p)
+        · intro _ mem
+          have aux1 := Subtype.mk_lt_mk.mp mem.1.1
+          have aux2 := Subtype.mk_lt_mk.mp mem.1.2
+          rw [Subtype.mk_eq_mk]
+          rw [hq] at aux2 ⊢
+          exact ((succ_le_iff.mpr aux1).antisymm (le_of_lt_succ aux2)).symm
+      obtain ⟨l, hl⟩ := exists_Ioc_subset_of_mem_nhds hu ⟨0, plim.pos⟩
+      obtain ⟨x, hx⟩ := h (Ioo ⟨l, hl.1.trans hpo⟩ ⟨p + 1, ho⟩) (Ioo_mem_nhds hl.1 (lt_add_one p))
+      use x
+      exact ⟨⟨hl.2 ⟨hx.1.1.1, lt_succ_iff.mp hx.1.1.2⟩, hx.1.2⟩, fun h ↦ hx.2 (SetCoe.ext h)⟩
+    have hp : o = p + 1 := (le_succ_iff_eq_or_le.mp (le_of_not_lt ho)).resolve_right
+      (not_le_of_lt hpo)
+    have ppos : p ≠ 0 := by
+      rintro rfl
+      obtain ⟨x, hx⟩ := h Set.univ univ_mem
+      have : ↑x < o := x.2
+      simp_rw [hp, zero_add, lt_one_iff_zero] at this
+      exact hx.2 (SetCoe.ext this)
+    obtain ⟨l, hl⟩ := exists_Ioc_subset_of_mem_nhds hu ⟨0, Ordinal.pos_iff_ne_zero.mpr ppos⟩
+    obtain ⟨x, hx⟩ := h (Ioi ⟨l, hl.1.trans hpo⟩) (Ioi_mem_nhds hl.1)
+    use x
+    refine ⟨⟨hl.2 ⟨hx.1.1, ?_⟩, hx.1.2⟩, fun h ↦ hx.2 (SetCoe.ext h)⟩
+    rw [← lt_add_one_iff, ← hp]
+    exact x.2
+
+theorem isClosedBelow_iff (S : Set Ordinal) (o : Ordinal) : IsClosedBelow S o ↔
+    ∀ p < o, IsAcc p S → p ∈ S := by
+  dsimp [IsClosedBelow]
+  constructor
+  · intro h p plto hp
+    have : AccPt ⟨p, plto⟩ (𝓟 (Iio o ↓∩ S)) := (accPt_subtype _ _).mp hp
+    rw [isClosed_iff_clusterPt] at h
+    exact h ⟨p, plto⟩ this.clusterPt
+  · intro h
+    rw [isClosed_iff_clusterPt]
+    intro r hr
+    match clusterPt_principal hr with
+    | .inl h => exact h
+    | .inr h' => exact h r.1 r.2 <| (accPt_subtype _ _).mpr h'
+
+theorem IsClosedBelow.forall_lt {S : Set Ordinal} {o : Ordinal} (h : IsClosedBelow S o) :
+    ∀ p < o, IsAcc p S → p ∈ S := (isClosedBelow_iff _ _).mp h
```

Comment:
I think this should work
```suggestion
alias ⟨IsClosedBelow.forall_lt, _⟩ := isClosedBelow_iff
```

Reply:
It does but then you need an extra parentheses around `.forall_lt`, ie `(h C CmemS).forall_lt p plto ...` -> `((h C CmemS).forall_lt) p plto ...`

Reply:
That doesn't seem right, function application doesn't need parentheses.

Reply:
Alright, I think i figured it out. Creating the alias automatically turned some arguments from implicit to explicit, arguments like `S` that are part of the definition of `IsClosedBelow`. So then `(h C CmemS).forall_lt p` didn't typecheck because `(h C CmemS).forall_lt` expected the now explicit arguments `S` and `o`. But apparantly surrounding an expressing like that with parentheses forces lean to fill in these inputs, so `((h C CmemS).forall_lt) = (h C CmemS).forall_lt C o`.

Also I'm now seeing that this problem just went away hahaha. I must have copied something wrong and somehow made the alias have explicit inputs or something.